### ISSUE TITLE
SQL data sources doesn't support alert notification templating using tags/labels

### ISSUE THREAD

**Issue by dpasqualin** [grafana/grafana#32899]
**What happened**:

I'm trying to set up alert notification with templating as specified in the [documentation](https://grafana.com/docs/grafana/latest/alerting/add-notification-template/), using PostgreSQL as data source (without TimescaleDB) and Prometheus AlertManager as notification channel, but I can't get it to display the actual values instead of the label string (e.g: `${instance}`).

**What you expected to happen**:

I expect to see in my alert system the values returned form the database instead of the string `${Label}` as defined in the alert window.

**How to reproduce it (as minimally and precisely as possible)**:

With the query below I tried to use the `${metric}` in the message field.
```
SELECT
  view_date_hour as "time",
  platform AS metric,
  AVG(buffering_time) AS "Buffering Time"
FROM
  ...
GROUP BY
  1,2
```

And I also tried [a possibly non-sense solution] to simulate the dimensions by manipulating the returning string, and then using `${key}` and `${value}` in the message field.

```
SELECT
  view_date_hour as "time",
  '{key="platform", value="' || platform || '"}' AS metric,
  AVG(buffering_time) AS "Buffering Time"
FROM
  ...
GROUP BY
  1,2
``` 
In both cases I get the string `${...}` instead of the actual values on AlertManager. I'm not sure if this is a bug, or templating with SQL databases would be a feature request, but either way it would be nice to have some info about it in the documentation.

Thank you very much.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.4.3, also tested on 7.5.3.
- Data source type & version: PostgreSQL 12

**Comment by hugohaggmark** [comment]
Adding this to @grafana/backend-platform board.

**Comment by marefr** [comment]
Seems like this is not supported for Postgres, MySQL and MSSQL since it doesn't set `tags` in the response.

For example the Prometheus data source set the tags in the response since it has label key/value pairs, ref https://github.com/grafana/grafana/blob/d2bdb4ed41247e7ad24dbfb6d13f2d5faf7c381c/pkg/tsdb/prometheus/prometheus.go#L209-L211

But for Postgres, MySQL and MSSQL we don't set any tags, ref
https://github.com/grafana/grafana/blob/d2bdb4ed41247e7ad24dbfb6d13f2d5faf7c381c/pkg/tsdb/sqleng/sql_engine.go#L431

Not sure what to provide in tags though since there's no key/value pairs, perhaps `metric` as key and `<metric value>` as value? 

We currently converting Postgres, MySQL and MSSQL to return data frames and that would perhaps automatically fix this, but not sure. @ying-jeanne something to think about

**Comment by kylebrandt** [comment]
I believe with the conversion to data frames this will eventually be supported.

In short, any string columns included in the time series will become labels, with the column name for a string column being the label/tag key, and the value of the column become the label/tag value.

**Comment by dpasqualin** [comment]
Thank you @kylebrandt @marefr and @hugohaggmark for the quick response, looking forward for the fix :)

**Comment by lux4rd0** [comment]
This seems to be the same kind of expectation for InfluxDB as well, correct? I thought I'd be able to use any of the `$tag_` variables, but none seem to come across. I can only get this to work when using Prometheus data sources. Are there examples of how this can be done with other data sources?

Example:

`SELECT mean("usage_idle") * -1 + 100 AS "alias" FROM "cpu" WHERE ("os" = 'linux') AND $timeFilter GROUP BY time($__interval), "host"`

And using `$host` gives me no variable...

**Comment by FireEmerald** [comment]
The feature _"SQL data sources: Convert to data frames"_ (#32257) seems to be merged. So whats the status of this request? I couldn't find any documentation about _"string columns included in the time series will become labels"_ for sql data sources. Which data source does support labels, which doesn't - any page in docu to refer to?

For me it seems like the new alerting in grafana 8 heavily relies on the usage of labels, but to use them they must be provided by the data source. Thus if your data source does not have out-of-the-box labels the new alerting is not really useable, or am i wrong?

**Comment by FireEmerald** [comment]
@kylebrandt I think the conversion to data frames is done(?) but still i couldn't find a way to write any query for example against a postgres db which detected the additional columns as key/value pairs. Am i missing something or is it still not possible?

**Comment by jessabe** [comment]
Closing this as it should be fixed as of 8.0. Please let us know though and we'll happily look into things again!
